From 2cd802030ab59787a34c0f6684c16848befabafa Mon Sep 17 00:00:00 2001
From: Imre Kis <imre.kis@arm.com>
Date: Wed, 15 Jun 2022 12:47:37 +0200
Subject: [PATCH 17/24] Move libsp mocks into separate component

Enable deployments to include libsp mocks in tests by simply adding
the newly created libsp mock component.

Signed-off-by: Imre Kis <imre.kis@arm.com>
Change-Id: I40805fd49362c6cc71b5b34f9ba888d27ce01ed8

Upstream-Status: Pending [In review]
Signed-off-by: Anton Antonov <Anton.Antonov@arm.com>

---
 .../messaging/ffa/libsp/mock/component.cmake  | 27 ++++++++++
 .../ffa/libsp/{test => mock}/mock_assert.cpp  |  0
 .../ffa/libsp/{test => mock}/mock_assert.h    |  0
 .../ffa/libsp/{test => mock}/mock_ffa_api.cpp |  0
 .../ffa/libsp/{test => mock}/mock_ffa_api.h   |  0
 .../{test => mock}/mock_ffa_internal_api.cpp  |  0
 .../{test => mock}/mock_ffa_internal_api.h    |  0
 .../ffa/libsp/{test => mock}/mock_sp_rxtx.cpp |  0
 .../ffa/libsp/{test => mock}/mock_sp_rxtx.h   |  0
 .../{ => mock}/test/test_mock_assert.cpp      |  0
 .../{ => mock}/test/test_mock_ffa_api.cpp     |  0
 .../test/test_mock_ffa_internal_api.cpp       |  0
 .../{ => mock}/test/test_mock_sp_rxtx.cpp     |  0
 components/messaging/ffa/libsp/tests.cmake    | 51 +++++++++++--------
 .../mm_communicate/endpoint/sp/tests.cmake    |  6 +--
 .../frontend/mm_communicate/tests.cmake       |  6 +--
 16 files changed, 64 insertions(+), 26 deletions(-)
 create mode 100644 components/messaging/ffa/libsp/mock/component.cmake
 rename components/messaging/ffa/libsp/{test => mock}/mock_assert.cpp (100%)
 rename components/messaging/ffa/libsp/{test => mock}/mock_assert.h (100%)
 rename components/messaging/ffa/libsp/{test => mock}/mock_ffa_api.cpp (100%)
 rename components/messaging/ffa/libsp/{test => mock}/mock_ffa_api.h (100%)
 rename components/messaging/ffa/libsp/{test => mock}/mock_ffa_internal_api.cpp (100%)
 rename components/messaging/ffa/libsp/{test => mock}/mock_ffa_internal_api.h (100%)
 rename components/messaging/ffa/libsp/{test => mock}/mock_sp_rxtx.cpp (100%)
 rename components/messaging/ffa/libsp/{test => mock}/mock_sp_rxtx.h (100%)
 rename components/messaging/ffa/libsp/{ => mock}/test/test_mock_assert.cpp (100%)
 rename components/messaging/ffa/libsp/{ => mock}/test/test_mock_ffa_api.cpp (100%)
 rename components/messaging/ffa/libsp/{ => mock}/test/test_mock_ffa_internal_api.cpp (100%)
 rename components/messaging/ffa/libsp/{ => mock}/test/test_mock_sp_rxtx.cpp (100%)

diff --git a/components/messaging/ffa/libsp/mock/component.cmake b/components/messaging/ffa/libsp/mock/component.cmake
new file mode 100644
index 0000000..03b8006
--- /dev/null
+++ b/components/messaging/ffa/libsp/mock/component.cmake
@@ -0,0 +1,27 @@
+#-------------------------------------------------------------------------------
+# Copyright (c) 2022, Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+#-------------------------------------------------------------------------------
+if (NOT DEFINED TGT)
+	message(FATAL_ERROR "mandatory parameter TGT is not defined.")
+endif()
+
+target_sources(${TGT} PRIVATE
+	"${CMAKE_CURRENT_LIST_DIR}/mock_assert.cpp"
+	"${CMAKE_CURRENT_LIST_DIR}/mock_ffa_api.cpp"
+	"${CMAKE_CURRENT_LIST_DIR}/mock_ffa_internal_api.cpp"
+	"${CMAKE_CURRENT_LIST_DIR}/mock_sp_rxtx.cpp"
+	)
+
+target_include_directories(${TGT}
+	PUBLIC
+		${CMAKE_CURRENT_LIST_DIR}
+		${CMAKE_CURRENT_LIST_DIR}/../include
+)
+
+target_compile_definitions(${TGT}
+	PUBLIC
+		"ARM64=1"
+)
\ No newline at end of file
diff --git a/components/messaging/ffa/libsp/test/mock_assert.cpp b/components/messaging/ffa/libsp/mock/mock_assert.cpp
similarity index 100%
rename from components/messaging/ffa/libsp/test/mock_assert.cpp
rename to components/messaging/ffa/libsp/mock/mock_assert.cpp
diff --git a/components/messaging/ffa/libsp/test/mock_assert.h b/components/messaging/ffa/libsp/mock/mock_assert.h
similarity index 100%
rename from components/messaging/ffa/libsp/test/mock_assert.h
rename to components/messaging/ffa/libsp/mock/mock_assert.h
diff --git a/components/messaging/ffa/libsp/test/mock_ffa_api.cpp b/components/messaging/ffa/libsp/mock/mock_ffa_api.cpp
similarity index 100%
rename from components/messaging/ffa/libsp/test/mock_ffa_api.cpp
rename to components/messaging/ffa/libsp/mock/mock_ffa_api.cpp
diff --git a/components/messaging/ffa/libsp/test/mock_ffa_api.h b/components/messaging/ffa/libsp/mock/mock_ffa_api.h
similarity index 100%
rename from components/messaging/ffa/libsp/test/mock_ffa_api.h
rename to components/messaging/ffa/libsp/mock/mock_ffa_api.h
diff --git a/components/messaging/ffa/libsp/test/mock_ffa_internal_api.cpp b/components/messaging/ffa/libsp/mock/mock_ffa_internal_api.cpp
similarity index 100%
rename from components/messaging/ffa/libsp/test/mock_ffa_internal_api.cpp
rename to components/messaging/ffa/libsp/mock/mock_ffa_internal_api.cpp
diff --git a/components/messaging/ffa/libsp/test/mock_ffa_internal_api.h b/components/messaging/ffa/libsp/mock/mock_ffa_internal_api.h
similarity index 100%
rename from components/messaging/ffa/libsp/test/mock_ffa_internal_api.h
rename to components/messaging/ffa/libsp/mock/mock_ffa_internal_api.h
diff --git a/components/messaging/ffa/libsp/test/mock_sp_rxtx.cpp b/components/messaging/ffa/libsp/mock/mock_sp_rxtx.cpp
similarity index 100%
rename from components/messaging/ffa/libsp/test/mock_sp_rxtx.cpp
rename to components/messaging/ffa/libsp/mock/mock_sp_rxtx.cpp
diff --git a/components/messaging/ffa/libsp/test/mock_sp_rxtx.h b/components/messaging/ffa/libsp/mock/mock_sp_rxtx.h
similarity index 100%
rename from components/messaging/ffa/libsp/test/mock_sp_rxtx.h
rename to components/messaging/ffa/libsp/mock/mock_sp_rxtx.h
diff --git a/components/messaging/ffa/libsp/test/test_mock_assert.cpp b/components/messaging/ffa/libsp/mock/test/test_mock_assert.cpp
similarity index 100%
rename from components/messaging/ffa/libsp/test/test_mock_assert.cpp
rename to components/messaging/ffa/libsp/mock/test/test_mock_assert.cpp
diff --git a/components/messaging/ffa/libsp/test/test_mock_ffa_api.cpp b/components/messaging/ffa/libsp/mock/test/test_mock_ffa_api.cpp
similarity index 100%
rename from components/messaging/ffa/libsp/test/test_mock_ffa_api.cpp
rename to components/messaging/ffa/libsp/mock/test/test_mock_ffa_api.cpp
diff --git a/components/messaging/ffa/libsp/test/test_mock_ffa_internal_api.cpp b/components/messaging/ffa/libsp/mock/test/test_mock_ffa_internal_api.cpp
similarity index 100%
rename from components/messaging/ffa/libsp/test/test_mock_ffa_internal_api.cpp
rename to components/messaging/ffa/libsp/mock/test/test_mock_ffa_internal_api.cpp
diff --git a/components/messaging/ffa/libsp/test/test_mock_sp_rxtx.cpp b/components/messaging/ffa/libsp/mock/test/test_mock_sp_rxtx.cpp
similarity index 100%
rename from components/messaging/ffa/libsp/test/test_mock_sp_rxtx.cpp
rename to components/messaging/ffa/libsp/mock/test/test_mock_sp_rxtx.cpp
diff --git a/components/messaging/ffa/libsp/tests.cmake b/components/messaging/ffa/libsp/tests.cmake
index d851442..296ae46 100644
--- a/components/messaging/ffa/libsp/tests.cmake
+++ b/components/messaging/ffa/libsp/tests.cmake
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2020-2021, Arm Limited. All rights reserved.
+# Copyright (c) 2020-2022, Arm Limited. All rights reserved.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 #
@@ -9,10 +9,11 @@ include(UnitTest)
 unit_test_add_suite(
 	NAME libsp_mock_assert
 	SOURCES
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_assert.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/test_mock_assert.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_assert.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/test/test_mock_assert.cpp
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -21,10 +22,11 @@ unit_test_add_suite(
 unit_test_add_suite(
 	NAME libsp_mock_ffa_internal_api
 	SOURCES
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_ffa_internal_api.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/test_mock_ffa_internal_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_ffa_internal_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/test/test_mock_ffa_internal_api.cpp
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -35,12 +37,13 @@ unit_test_add_suite(
 	SOURCES
 		${CMAKE_CURRENT_LIST_DIR}/test/test_ffa_api.cpp
 		${CMAKE_CURRENT_LIST_DIR}/test/test_ffa_memory_descriptors.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_ffa_internal_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_ffa_internal_api.cpp
 		${CMAKE_CURRENT_LIST_DIR}/ffa.c
 		${CMAKE_CURRENT_LIST_DIR}/ffa_memory_descriptors.c
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_assert.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_assert.cpp
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -49,10 +52,11 @@ unit_test_add_suite(
 unit_test_add_suite(
 	NAME libsp_mock_ffa_api
 	SOURCES
-		${CMAKE_CURRENT_LIST_DIR}/test/test_mock_ffa_api.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_ffa_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/test/test_mock_ffa_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_ffa_api.cpp
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -62,10 +66,11 @@ unit_test_add_suite(
 	NAME libsp_sp_rxtx
 	SOURCES
 		${CMAKE_CURRENT_LIST_DIR}/test/test_sp_rxtx.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_ffa_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_ffa_api.cpp
 		${CMAKE_CURRENT_LIST_DIR}/sp_rxtx.c
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -74,10 +79,11 @@ unit_test_add_suite(
 unit_test_add_suite(
 	NAME libsp_mock_sp_rxtx
 	SOURCES
-		${CMAKE_CURRENT_LIST_DIR}/test/test_mock_sp_rxtx.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_sp_rxtx.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/test/test_mock_sp_rxtx.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_sp_rxtx.cpp
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -88,10 +94,11 @@ unit_test_add_suite(
 	SOURCES
 		${CMAKE_CURRENT_LIST_DIR}/test/test_sp_discovery.cpp
 		${CMAKE_CURRENT_LIST_DIR}/sp_discovery.c
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_ffa_api.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_sp_rxtx.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_ffa_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_sp_rxtx.cpp
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -103,11 +110,12 @@ unit_test_add_suite(
 		${CMAKE_CURRENT_LIST_DIR}/test/test_sp_memory_management.cpp
 		${CMAKE_CURRENT_LIST_DIR}/sp_memory_management.c
 		${CMAKE_CURRENT_LIST_DIR}/ffa_memory_descriptors.c
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_assert.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_ffa_api.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_sp_rxtx.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_assert.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_ffa_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_sp_rxtx.cpp
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -119,9 +127,10 @@ unit_test_add_suite(
 		${CMAKE_CURRENT_LIST_DIR}/test/test_sp_memory_management_internals.cpp
 		${CMAKE_CURRENT_LIST_DIR}/test/sp_memory_management_internals.yml
 		${CMAKE_CURRENT_LIST_DIR}/ffa_memory_descriptors.c
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_assert.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_assert.cpp
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -131,10 +140,11 @@ unit_test_add_suite(
 	NAME libsp_sp_messaging
 	SOURCES
 		${CMAKE_CURRENT_LIST_DIR}/test/test_sp_messaging.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_ffa_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_ffa_api.cpp
 		${CMAKE_CURRENT_LIST_DIR}/sp_messaging.c
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
@@ -144,11 +154,12 @@ unit_test_add_suite(
 	NAME libsp_sp_messaging_with_routing_extension
 	SOURCES
 		${CMAKE_CURRENT_LIST_DIR}/test/test_sp_messaging.cpp
-		${CMAKE_CURRENT_LIST_DIR}/test/mock_ffa_api.cpp
+		${CMAKE_CURRENT_LIST_DIR}/mock/mock_ffa_api.cpp
 		${CMAKE_CURRENT_LIST_DIR}/sp_messaging.c
 		${CMAKE_CURRENT_LIST_DIR}/ffa_direct_msg_routing_extension.c
 	INCLUDE_DIRECTORIES
 		${CMAKE_CURRENT_LIST_DIR}/include/
+		${CMAKE_CURRENT_LIST_DIR}/mock
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 	COMPILE_DEFINITIONS
 		-DARM64
diff --git a/components/rpc/mm_communicate/endpoint/sp/tests.cmake b/components/rpc/mm_communicate/endpoint/sp/tests.cmake
index 318f14d..c68a0c7 100644
--- a/components/rpc/mm_communicate/endpoint/sp/tests.cmake
+++ b/components/rpc/mm_communicate/endpoint/sp/tests.cmake
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2021, Arm Limited. All rights reserved.
+# Copyright (c) 2021-2022, Arm Limited. All rights reserved.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 #
@@ -13,12 +13,12 @@ unit_test_add_suite(
 		${CMAKE_CURRENT_LIST_DIR}/test/test_mm_communicate_call_ep.cpp
 		${CMAKE_CURRENT_LIST_DIR}/test/mock_mm_service.cpp
 		${CMAKE_CURRENT_LIST_DIR}/test/test_mock_mm_service.cpp
-		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/test/mock_assert.cpp
+		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/mock/mock_assert.cpp
 	INCLUDE_DIRECTORIES
 		${UNIT_TEST_PROJECT_PATH}
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/include
-		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/test
+		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/mock
 		${UNIT_TEST_PROJECT_PATH}/components/rpc/common/interface
 	COMPILE_DEFINITIONS
 		-DARM64
diff --git a/components/service/smm_variable/frontend/mm_communicate/tests.cmake b/components/service/smm_variable/frontend/mm_communicate/tests.cmake
index d1f930c..50b0b9a 100644
--- a/components/service/smm_variable/frontend/mm_communicate/tests.cmake
+++ b/components/service/smm_variable/frontend/mm_communicate/tests.cmake
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2021, Arm Limited. All rights reserved.
+# Copyright (c) 2021-2022, Arm Limited. All rights reserved.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 #
@@ -12,13 +12,13 @@ unit_test_add_suite(
 		${CMAKE_CURRENT_LIST_DIR}/smm_variable_mm_service.c
 		${CMAKE_CURRENT_LIST_DIR}/test/test_smm_variable_mm_service.cpp
 		${UNIT_TEST_PROJECT_PATH}/components/rpc/common/test/mock_rpc_interface.cpp
-		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/test/mock_assert.cpp
+		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/mock/mock_assert.cpp
 	INCLUDE_DIRECTORIES
 		${UNIT_TEST_PROJECT_PATH}
 		${UNIT_TEST_PROJECT_PATH}/components/rpc/mm_communicate/endpoint/sp
 		${UNIT_TEST_PROJECT_PATH}/components/common/utils/include
 		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/include
-		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/test
+		${UNIT_TEST_PROJECT_PATH}/components/messaging/ffa/libsp/mock
 		${UNIT_TEST_PROJECT_PATH}/components/rpc/common/interface
 	COMPILE_DEFINITIONS
 		-DARM64
-- 
2.17.1

